hack/generator/pkg/astmodel/armconversion/shared.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package armconversion
+
+import (
+	"fmt"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"go/ast"
+	"sync"
+)
+
+type conversionBuilder struct {
+	receiverIdent              *ast.Ident
+	receiverTypeExpr           ast.Expr
+	armTypeIdent               *ast.Ident
+	codeGenerationContext      *astmodel.CodeGenerationContext
+	idFactory                  astmodel.IdentifierFactory
+	isResource                 bool
+	methodName                 string
+	kubeType                   *astmodel.ObjectType
+	armType                    *astmodel.ObjectType
+	propertyConversionHandlers []propertyConversionHandler
+}
+
+func (builder conversionBuilder) propertyConversionHandler(
+	toProp *astmodel.PropertyDefinition,
+	fromType *astmodel.ObjectType) []ast.Stmt {
+
+	for _, conversionHandler := range builder.propertyConversionHandlers {
+		matched, matchedProperty := conversionHandler.finder(toProp, fromType)
+		if matched {
+			return conversionHandler.matchHandler(toProp, matchedProperty)
+		}
+	}
+
+	panic(fmt.Sprintf("No property found for %s", toProp.PropertyName()))
+}
+
+type propertyMatchHandlerFunc = func(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []ast.Stmt
+
+type propertyConversionHandler struct {
+	finder       func(toProp *astmodel.PropertyDefinition, fromType *astmodel.ObjectType) (bool, *astmodel.PropertyDefinition)
```

Comment:
Yeah I agree `finder` isn't a good name.

Since we have `matchHandler` below we could just call it `match`? I'm also okay with `selector`. We _are_ doing basically a LINQ `select` here (mapping from a collection of property tuples to conversions between the tuple elements). Thoughts?

Reply:
based on @babbageclunk's suggestion, I've changed this so that it's just a function rather than 2, so this doesn't apply anymore. Closing.